### ISSUE TITLE
Error: Access is allowed from event dispatch thread with IW lock only.

### ISSUE THREAD

**Issue by Trinsdar** [Almighty-Alpaca/JetBrains-Discord-Integration#105]
Using 1.4.2, intellj idea reported this upon start(I think): 
https://hastebin.com/erubobagoh.sql
Using ubuntu 20.04

<details>
<summary>Exception</summary>

```java
Access is allowed from event dispatch thread with IW lock only.

com.intellij.openapi.diagnostic.RuntimeExceptionWithAttachments: EventQueue.isDispatchThread()=false Toolkit.getEventQueue()=com.intellij.ide.IdeEventQueue@8da674b
	Current thread: Thread[DefaultDispatcher-worker-2,4,Idea Thread Group] 1610569668
	SystemEventQueueThread: Thread[AWT-EventQueue-0,6,Idea Thread Group] 233205247
	at com.intellij.openapi.application.impl.ApplicationImpl.assertIsDispatchThread(ApplicationImpl.java:1057)
	at com.intellij.openapi.application.impl.ApplicationImpl.assertIsDispatchThread(ApplicationImpl.java:1039)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.getSelectedEditorWithProvider(FileEditorManagerImpl.java:1283)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.getSelectedEditor(FileEditorManagerImpl.java:1275)
	at com.intellij.openapi.fileEditor.FileEditorManager.getSelectedEditor(FileEditorManager.java:109)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.getSelectedEditor(FileEditorManagerImpl.java:1269)
	at com.almightyalpaca.jetbrains.plugins.discord.plugin.data.DataService.getDataInternal(DataService.kt:71)
	at com.almightyalpaca.jetbrains.plugins.discord.plugin.data.DataService.getData(DataService.kt:49)
	at com.almightyalpaca.jetbrains.plugins.discord.plugin.render.RenderService$render$2.invokeSuspend(RenderService.kt:63)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineSchedule
```

</details>

**Comment by InfRandomness** [comment]
The hastebin link seems to have no content, can you send another valid one ?

**Comment by InfRandomness** [comment]
The link seems to work now

**Comment by dwcaesar** [comment]
This exception can also occur after running for a while.
It doesn't seem to be dependend on the open project, programming language or using multiple instances at the same time.

<details>
<summary>Access is allowed from event dispatch thread with IW lock only.</summary>

>com.intellij.openapi.diagnostic.RuntimeExceptionWithAttachments: EventQueue.isDispatchThread()=false >Toolkit.getEventQueue()=com.intellij.ide.IdeEventQueue@58552a9
>Current thread: Thread[DefaultDispatcher-worker-3,4,Idea Thread Group] 1393436708
>SystemEventQueueThread: Thread[AWT-EventQueue-0,6,Idea Thread Group] 2030475492
>	at com.intellij.openapi.application.impl.ApplicationImpl.assertIsDispatchThread(ApplicationImpl.java:1061)
>	at com.intellij.openapi.application.impl.ApplicationImpl.assertIsDispatchThread(ApplicationImpl.java:1043)
>	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.getSelectedEditorWithProvider(FileEditorManagerImpl.java:1252)
>	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.getSelectedEditor(FileEditorManagerImpl.java:1245)
>	at com.intellij.openapi.fileEditor.FileEditorManager.getSelectedEditor(FileEditorManager.java:126)
>	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.getSelectedEditor(FileEditorManagerImpl.java:1240)
>	at com.almightyalpaca.jetbrains.plugins.discord.plugin.data.DataService.getDataInternal(DataService.kt:71)
>	at com.almightyalpaca.jetbrains.plugins.discord.plugin.data.DataService.getData(DataService.kt:49)
>	at com.almightyalpaca.jetbrains.plugins.discord.plugin.render.RenderService$render$2.invokeSuspend(RenderService.kt:63)
>	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
>	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
>	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
>	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)
>	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
>	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
</details>

<details>
<summary>About Dialog from IntelliJ</summary>

> IntelliJ IDEA 2020.2 (Ultimate Edition)
> Build #IU-202.6397.94, built on July 27, 2020
> Licensed to me
> Subscription is active until January 12, 2021
> Runtime version: 11.0.7+10-b944.20 amd64
> VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
> Linux 5.4.0-42-generic
> GC: ParNew, ConcurrentMarkSweep
> Memory: 1954M
> Cores: 4
> Registry: documentation.show.toolbar=true, analyze.exceptions.on.the.fly=true, search.everywhere.settings=true, ide.balloon.shadow.size=0
> Non-Bundled Plugins: YAML/Ansible support, com.jetbrains.ChooseRuntime, com.jetbrains.plugins.ini4idea, net.seesharpsoft.intellij.plugins.csv, nl.jworks.intellij.bootstrap3, org.intellij.RegexpTester, com.almightyalpaca.intellij.plugins.discord, com.intellij.plugin.adernov.powershell, com.perl5, com.intellij.plugins.html.instantEditing, com.jetbrains.php, com.jetbrains.php.blade, de.espend.idea.laravel, Pythonid, net.ashald.envfile, ru.adelf.idea.dotenv
> Current Desktop: KDE
</details>

**Comment by Almighty-Alpaca** [comment]
A fix for this has just been released with v1.5.0.
